### PR TITLE
Fixed issue where null JWT sent to Core on startup

### PR DESCRIPTION
**Pull request by thomasm-ttd**
Added unit test
Tested in unit tests locally
Tested in E2E tests locally
Tested in a branch deploy of both Operator and Optout

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
     <groupId>com.uid2</groupId>
     <artifactId>uid2-shared</artifactId>
-    <version>6.0.10-208e0b52b2</version>
+    <version>6.0.14-SNAPSHOT</version>
```
Comment (gmsdelmundo) [inline pom.xml:9]: Don't forget to change this back

File: src/test/java/com/uid2/shared/attest/UidCoreClientTest.java [hunk 2]
```diff
@@ -59,6 +55,37 @@ public void Download_Succeed_RequestSentWithExpectedParameters() throws IOExcept
         when(mockHttpClient.get("https://download", expectedHeaders)).thenReturn(mockHttpResponse);
 
         uidCoreClient.download("https://download");
+        verify(mockAttestationTokenRetriever, times(1)).attest();
+        verify(mockHttpClient, times(1)).get("https://download", expectedHeaders);
+    }
+
+    @Test
+    public void Download_With_Attest_Uses_JWT_Second_Call() throws IOException, CloudStorageException, AttestationTokenRetrieverException {
```
Comment (gmsdelmundo) [inline src/test/java/com/uid2/shared/attest/UidCoreClientTest.java:63]: Can the method name be `DownloadWithAttest_UsesJWTSecondCall()`? Or something along those lines

File: src/test/java/com/uid2/shared/attest/UidCoreClientTest.java [hunk 3]
```diff
@@ -59,6 +55,37 @@ public void Download_Succeed_RequestSentWithExpectedParameters() throws IOExcept
         when(mockHttpClient.get("https://download", expectedHeaders)).thenReturn(mockHttpResponse);
 
         uidCoreClient.download("https://download");
+        verify(mockAttestationTokenRetriever, times(1)).attest();
```
Comment (gmsdelmundo) [inline src/test/java/com/uid2/shared/attest/UidCoreClientTest.java:58]: Can you wrap the `verify()`s under an `assertAll()`? Similar comment for the other test

File: src/main/java/com/uid2/shared/attest/UidCoreClient.java [hunk 4]
```diff
@@ -108,6 +106,9 @@ private InputStream getWithAttest(String path, String jwtToken) throws IOExcepti
         }
 
         String attestationToken = attestationTokenRetriever.getAttestationToken();
+        if (jwtToken == null || jwtToken.isEmpty()) {
```
Comment (gmsdelmundo) [inline src/main/java/com/uid2/shared/attest/UidCoreClient.java:109]: Would it be better to put this just after `attestationTokenRetriever.attest();`? `jwtToken` would only really be null if we haven't attested yet

i.e.

```
        if (!attestationTokenRetriever.attested()) {
            attestationTokenRetriever.attest();
            jwtToken = this.getJWT();
        }
```

File: pom.xml (hunk 1 above)
Comment (thomasm-ttd) [reply pom.xml:9]: This will get fixed by the build

File: src/test/java/com/uid2/shared/attest/UidCoreClientTest.java (hunk 2 above)
Comment (thomasm-ttd) [reply src/test/java/com/uid2/shared/attest/UidCoreClientTest.java]: Fixed

File: src/main/java/com/uid2/shared/attest/UidCoreClient.java (hunk 4 above)
Comment (thomasm-ttd) [reply src/main/java/com/uid2/shared/attest/UidCoreClient.java]: Moved the code, but still left the check. There are other reasons we might not have attested, such as core being unavailable

Comment (thomasm-ttd) [reply src/main/java/com/uid2/shared/attest/UidCoreClient.java]: Removed the parameter and call getJWT just before we need it

File: src/main/java/com/uid2/shared/attest/UidCoreClient.java [hunk 5]
```diff
@@ -137,8 +128,10 @@ private HttpResponse<String> sendHttpRequest(String path, String attestationToke
         if (attestationToken != null && !attestationToken.isBlank()) {
             headers.put(Const.Attestation.AttestationTokenHeader, attestationToken);
         }
-        if (attestationJWT != null && !attestationJWT.isBlank()) {
-            headers.put(Const.Attestation.AttestationJWTHeader, attestationJWT);
+
+        String jwtToken = this.getJWT();
+        if (jwtToken != null && !jwtToken.isBlank()) {
```
Comment (vishalegbert-ttd) [inline src/main/java/com/uid2/shared/attest/UidCoreClient.java:133]: Will the JWT be blank at this point? Should we log if it is?

Comment (thomasm-ttd) [reply src/main/java/com/uid2/shared/attest/UidCoreClient.java:133]: It shouldn't be, but good to log it.

**Pull request merged**